### ISSUE TITLE
bug: StreamingOnlyPolicy can cause aggressive memory eviction in small memory setup

### ISSUE THREAD

**Issue by hzxa21** [risingwavelabs/risingwave#8822]
### Describe the bug

Currently StreamingOnlyPolicy (the default memory control policy) triggers eviction based on 
- `total_compute_memory_bytes` = `cn_total_memory_bytes` - `storage_memory_bytes` - `(SYSTEM_RESERVED_MEMORY_MB << 20)`
- `jemalloc_allocated_mib` = all allocated memory including storage and compute memory usage
```rust
       let jemalloc_allocated_mib =
            advance_jemalloc_epoch(prev_memory_stats.jemalloc_allocated_mib);
        let stream_memory_threshold_graceful =
            (total_compute_memory_bytes as f64 * Self::STREAM_EVICTION_THRESHOLD_GRACEFUL) as usize;
        let stream_memory_threshold_aggressive = (total_compute_memory_bytes as f64
            * Self::STREAM_EVICTION_THRESHOLD_AGGRESSIVE)
            as usize;

        // Streaming memory control
        //
        // We calculate the watermark of the LRU cache, which provides hints for streaming executors
        // on cache eviction. Here we do the calculation based on jemalloc statistics.

        let (lru_watermark_step, lru_watermark_time_ms, lru_physical_now) = calculate_lru_watermark(
            jemalloc_allocated_mib,
            stream_memory_threshold_graceful,
            stream_memory_threshold_aggressive,
            barrier_interval_ms,
            prev_memory_stats,
        );
```

Since we calculate the eviction threshold using only the compute portion of the memory usage but compare it against the total allocated memory in CN, it can easily trigger aggresive memory eviction for operator cache when CN has small memory. In one of the deployment, we saw that a 4GB memory CN always evict all cached entries in all operator cache when storage takes ~1.6GB memory, causing very bad performance during MV creation.

### To Reproduce

- Start up a cluster with 4GB memory CN and default config.
- Run some worklaod.
- Monitor the LRU manager evcition.

**Comment by hzxa21** [comment]
related: #8696

**Comment by fuyufjh** [comment]
Closed by #8827
